### PR TITLE
work around GPhoto2 crashing dt by disabling device scans during imports

### PR DESCRIPTION
**Pull request by ralfbrown**
...and by slowing down the scan rate.

This commit changes the interval for the background scan for added/removed cameras while lighttable view is active from every 3.2 seconds to every 8 seconds.  In various circumstances, the code was using an 0.4 second interval for responsiveness; that is now 1.0 seconds.

In addition, the existing mechanism to suppress scans while importing from camera (using GPhoto2) is also used to suppress scans while the add-to-library/copy-and-import window is open.

These changes mitigate the reported crashes on MacOS Ventura caused by a broken dylib loader. (#13221)

A more complete fix would also disable the scan whenever the import module is collapsed or hidden (e.g. left panel collapsed).

See also #14314.

### CODE REVIEW

**Comment by TurboGit** [comment]
CI issue:

```
/home/runner/work/darktable/darktable/src/src/libs/import.c:1968:3: fatal error: must use 'struct' tag to refer to type 'dt_camctl_t'
  dt_camctl_t *camctl = (dt_camctl_t *)darktable.camctl;
  ^
  struct 
1 error generated.
```

File: src/common/camera_control.c [hunk 1]
```diff
@@ -1013,13 +1013,20 @@ void *dt_update_cameras_thread(void *ptr)
     if(camctl)
     {
       if((camctl->import_ui == FALSE) && (cv && (cv->view(cv) == DT_VIEW_LIGHTTABLE)))
+//TODO:  && import module is expanded and visible
```
Comment (TurboGit) [inline src/common/camera_control.c:1016]: Yes, probably a good idea indeed, so it should be something like:

`dtgtk_expander_get_expanded(<import_module>->expander);`

But right here it is far from trivial as we don't have the import lib module. One solution would be to scan all the libs from darktable.lib GList, looking for "import" and checking the expander. Not sure we want to do that again and again.

So maybe store in main darktable struct a boolean `import_expanded` which is true when import is expanded? Anyway, no clean solution...

**Comment by TurboGit** [comment]
Probably a missing include:

```
 /home/runner/work/darktable/darktable/src/src/libs/import.c: In function ‘_lib_import_from_callback’:
/home/runner/work/darktable/darktable/src/src/libs/import.c:1968:33: error: ‘dt_camctl_t’ undeclared (first use in this function)
 1968 |   struct dt_camctl_t *camctl = (dt_camctl_t *)darktable.camctl;
      |                                 ^~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
```

File: src/common/camera_control.c (hunk 1 above)
Comment (ralfbrown) [reply src/common/camera_control.c:1016]: Yeah, I figured we'll probably need to set up a variable that is updated by the import module itself as it goes in and out of visibility.

**Comment by ralfbrown** [comment]
Oh, needs to be a conditional compilation since libgphoto2 is an optional dependency.

File: src/common/camera_control.c (hunk 1 above)
Comment (TurboGit) [reply src/common/camera_control.c:1016]: Do you want to do that as part of this PR or another one?

Comment (ralfbrown) [reply src/common/camera_control.c:1016]: Let's do that separately.

**Review by TurboGit — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**